packages/modules/src/scoring/Scoring.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  SCORE_MODULUS,
+  IJournalContent,
+  IRatingInput,
+  CordAddress,
+} from '@cord.network/types'
+import { Crypto } from '@cord.network/utils'
+import {
+  SCORE_IDENT,
+  SCORE_PREFIX,
+  ScoreType,
+  IScoreAverageDetails,
+} from '@cord.network/types'
+import { Identifier, SDKErrors } from '@cord.network/utils'
+import { fetchScore } from './Scoring.chain'
+
+export function adjustAndRoundRating(rating: number): number {
+  rating = Math.round(rating * SCORE_MODULUS)
+  return rating
+}
+
+export function ComputeActualRating(rating: number): number {
+  return rating / SCORE_MODULUS
+}
+
+export function ComputeAverageRating(rating: number, count: number): number {
+  return rating / count
+}
+
+export function generateDigestFromJournalContent(
+  journalContent: IJournalContent
+) {
+  const digest = Crypto.hash(JSON.stringify(journalContent))
+  const hexDigest = Crypto.u8aToHex(digest)
+  return hexDigest
+}
+
+export function getUriForScore(journalContent: IJournalContent) {
+  const scoreDigest = generateDigestFromJournalContent(journalContent)
+  return Identifier.hashToUri(scoreDigest, SCORE_IDENT, SCORE_PREFIX)
+}
+
+export function fromScore(
+  journalContent: IJournalContent,
+  creator: CordAddress
+): IRatingInput {
+  journalContent.rating = adjustAndRoundRating(journalContent.rating)
+  const digest = generateDigestFromJournalContent(journalContent)
+  const ratingInput: IRatingInput = {
+    entry: journalContent,
+    digest: digest,
+    creator: creator,
+  }
+  return ratingInput
+}
+
+export async function fetchAverageScore(
+  entityUri: string,
+  scoreType: ScoreType
+): Promise<IScoreAverageDetails> {
+  const pertialEntityUri = entityUri.split('did:cord:').join('')
+  const decodedScoreEntry = await fetchScore(pertialEntityUri, scoreType)
+
+  const actualRating = ComputeActualRating(decodedScoreEntry.rating)
+  const averageRating = ComputeAverageRating(
+    actualRating,
+    decodedScoreEntry.count
+  )
+  return {
+    rating: actualRating,
+    count: decodedScoreEntry.count,
+    average: averageRating,
+  }
+}
+
+export function fromJournalContent(
+  journalContent: IJournalContent,
+  creator: CordAddress
+) {
+  verifyScoreStructure(journalContent)
+  const ratingInput = fromScore(journalContent, creator)
+  const entry = ratingInput.entry
+  const scoreIdentifier = getUriForScore(entry)
+  return {
+    ratingInput,
+    scoreIdentifier,
+  }
+}
+
+export function verifyScoreStructure(input: IJournalContent) {
+  if (!input.collector) {
+    throw new SDKErrors.ScoreCollectorMissingError()
+  }
+  if (!input.entity) {
+    throw new SDKErrors.ScoreEntityMissingError()
+  }
+  if (!input.tid) {
+    throw new SDKErrors.ScoreTidMissingError()
+  }
+  if (!input.entry_type) {
+    throw new SDKErrors.ScoreEntryTypeMissingError()
+  }
+  if (!input.count) {
+    throw new SDKErrors.ScoreCountMissingError()
+  }
+  if (!input.rating) {
```

Comment:
Should not be more that the max value. If, per rating entry max value is five, this entry should be within the limit of count * max_value

Reply:
@smohan-dw  consider this scenario - count 3, score 15 ( individual entries could have been 6,4 5)
so count * max_value will be 15, it'll pass the validation, but in reality one of the entries '6' is beyond max, so I don't think we justify this validation - right?
I think our validation here will make sense only if we were to get the individual ratings..

Reply:
Correct, managing the correctness of the individual entries is the responsibility of the rating provider. The SDK should make sure the max_value checks are in place. For e.g. an entry with a count of 10 should not have a total value exceeding 50.